### DISCUSSION TITLE
Restore originals from thumbnails?

### DISCUSSION THREAD

**Discussion by dessalines** [photoprism/photoprism#2432] [Q&A] [answered]
Somehow in an upgrade, photoprism deleted my entire originals folder :(

Luckily, I was able to save a backup of the `.photoprism` folder, which includes all the thumbnails, including high-res ones, as well as all the yml files. Unfortunately the thumbnails don't have the original exif data, so these are all completely unorganized.

After an upgrade to the newest photoprism, it now sees the thumbnails, but when I click on an album, it says "Cannot find".

Is there any way I can restore this data some way? Ideally I just want to restore dates on the exif data for the thumbnails folder, then I can remove the unecessary thumbnails, extract them to a new folder, and use it as my new "originals".

Has anyone had this problem before? How did you solve it?

**Comment by lastzero** [comment]
More likely, the folders are/were mounted incorrectly or not mounted at all, so the files were stored in the Docker container and not on the host?

We kindly ask you not to report bugs via GitHub Issues unless you are certain to have found a fully reproducible and previously unreported issue that must be fixed directly in the app. [Contact us](https://photoprism.app/contact) or [a community member](https://link.photoprism.app/discussions) if you need help, it could be a local configuration problem, or a misunderstanding in how the software works.

**Reply by dessalines** [reply]
I think unfortunately I'll have to write a script to rewrite the exif data, luckily it looks like it was preserved in `.json` files: 

```
cat cache/json/c/0/3/c03e484261bbdddbb81f6a703e8b5adf4b8b5bac_exiftool.json                                                         [{
  "SourceFile": "/photoprism/originals/2009-11-27/004.JPG",                                                                             "ExifToolVersion": 12.16,                                                                                                             "FileName": "004.JPG",                                                                                                                "Directory": "/photoprism/originals/2009-11-27",
  "FileSize": "269 KiB",                                                                                                                "FileModifyDate": "2016:05:23 00:14:46+00:00",
  "FileAccessDate": "2021:09:05 16:28:20+00:00",
  "FileInodeChangeDate": "2021:08:27 20:09:23+00:00",
  "FilePermissions": "rwxr-xr-x",
  "FileType": "JPEG",
  "FileTypeExtension": "jpg",
  "MIMEType": "image/jpeg",
```

**Comment by dessalines** [comment, marked as answer]
Okay, I've written a script to be able to do this. Maybe someone else will find it handy at some point. Hopefully the github search works with discussions too.

https://github.com/dessalines/restore_photoprism_originals

**Comment by Pnikanti** [comment]
Apologies for necrobumping. I lost 30k worth of pictures and videos (over the span of 2013 - 2024).. by my own mistake. I'm glad the Photoprism cache exists and your tool seems to work.. currently running it and restoring the pictures. LIFE SAVER! ❤️

**Reply by dessalines** [reply]
No probs :)
